README.md

```diff
@@ -207,23 +220,273 @@ More information can be found [here](http://www.diveintopython3.net/strings.html
+                module = "synapseutils",
+                modifyFunctions = selectFunctions)
+```
+
+And in your `.onLoad`, update `generateRWrappers` as following:
+```r
+generateRWrappers(pyPkg = "synapseutils",
+                  module = "synapseutils",
+                  setGenericCallback = callback,
+                  modifyFunctions = selectFunctions)
+```
+
+### Expose a subset of classes within a Python module
+
+The `synapseclient` module in `synapsePythonClient` has the following structure:
+```
+    module: synapseclient
+        module: client
+            class: Synapse
+                function: get
+                function: login
+        module: entity
+            class: Entity
+                function: get
+                function: privateGet
+            class: File
+                function: get
+                function: privateGet
+            class: Folder
+                function: get
+                function: privateGet
+```
+
+Now let's try a more complicated example where we want to expose the following functions and classes from `synapseclient.entity` module:
+```
+  class: File
+      function: get
+  class: Folder
+      function: get
+```
+
+Note that we do not want the following from the `synapseclient.entity` module:
+* `function: privateGet` from any of the classes
+* `class: Entity`
+
+First, define your `selectClasses` in the shared .R file:
+```r
+methodsToOmit <- "privateGet"
+classToSkip <- "Entity"
+selectClasses <- function(class) {
+    if (any(class$name == classToSkip)) {
+        return(NULL)
+    }
+    if (!is.null(class$methods)) {
+        culledMethods <- lapply(X = class$methods, function(x) {
+            if (any(x$name == methodsToOmit)) NULL else x;
+        })
+        # Now remove the nulls
+        nullIndices <- sapply(culledMethods, is.null)
+        if (any(nullIndices)) {
+            class$methods <- culledMethods[-which(nullIndices)]
+        }
+    }
+    class
+}
+```
+
+Then call `generateRdFiles` and `generateRWrappers` as following:
+```r
+generateRdFiles(srcRootDir,
+                pyPkg = "synapseclient",
+                module = "synapseclient.entity",
+                modifyClasses = selectClasses)
+```
+```r
+generateRWrappers(pyPkg = "synapseclient",
+                  module = "synapseclient.entity",
+                  setGenericCallback = callback,
+                  modifyClasses = selectClasses)
+```
+
+### Expose functions within a singleton object
+
+In rare cases, we want to expose Python functions under `synapseclient.client.Synapse` without exposing the `Synapse` object itself.
+
+In a Python session, we would do the following:
+```
+  syn = synapseclient.Synapse()
+  syn.login()
+  syn.get()
+```
+
+And in R, users would not need to know about the `Synapse` object:
+```
+  synLogin()
+  synGet()
+```
+
+Note that:
+* all function calls in R access the same underlying Python object `Synapse`
+* the functions name in R has a `syn` prefix
```

Comment:
In the following example, the function names in R will be prepended with a `syn` prefix.